app/gui/src/controller/searcher.rs

```diff
@@ -845,6 +894,8 @@ impl Searcher {
         let mut graph_info = GraphInfo::from_definition(graph_definition.item);
         graph_info.add_node(&node, LocationHint::End)?;
         module.ast = module.ast.set_traversing(&graph_definition.crumbs, graph_info.ast())?;
+        let position =
+            self.graph.graph().node(self.mode.node_id())?.metadata.and_then(|md| md.position);
         let metadata = NodeMetadata { position, ..default() };
```

Comment:
When adding an example, we still create a new node. Are we sure the old one is removed? Maybe we can just update the existing node expression instead of adding a new one?

Reply:
By default, the newly created node is cleared unless, we prevent that by calling `prevent_revert` on the edit guard. 
The only case that could result in a weird state is, if we edit a node and choose an example. Then the original node would be reverted to pre-edit and the example is inserted. But I think examples are only available for empty prompts and are not meant to be used with other nodes (at the moment).